cmd/containerd/command/main.go

```diff
@@ -322,6 +322,13 @@ func applyFlags(context *cli.Context, config *srvconfig.Config) error {
 		},
 	} {
 		if s := context.GlobalString(v.name); s != "" {
+			if v.name == "root" || v.name == "state" {
+				absPath, err := filepath.Abs(s)
+				if err != nil {
+					return err
+				}
+				*v.d = absPath
+			}
 			*v.d = s
```

Comment:
Else?

Reply:
Initially, I thought it's simpler to convert all paths to absolute. But the failed e2e made me think otherwise. That e2e seems broken so let me revert back to initial code.